src/groups/wrappers/z_bmq/z_bmqa/z_bmqa_message.cpp

```diff
@@ -0,0 +1,290 @@
+    using namespace BloombergLP;
+    const bmqa::Message* message_p = reinterpret_cast<const bmqa::Message*>(
+        message_obj);
+    *subscription_obj = reinterpret_cast<const z_bmqt_SubscriptionHandle*>(
+        &(message_p->subscriptionHandle()));
+
+    return 0;
+}
+
+z_bmqt_CompressionAlgorithmType::Enum
+z_bmqa_Message__compressionAlgorithmType(const z_bmqa_Message* message_obj)
+{
+    using namespace BloombergLP;
+    const bmqa::Message* message_p = reinterpret_cast<const bmqa::Message*>(
+        message_obj);
+
+    return static_cast<z_bmqt_CompressionAlgorithmType::Enum>(
+        message_p->compressionAlgorithmType());
+}
+
+#ifdef BMQ_ENABLE_MSG_GROUPID
+
+const char* z_bmqa_Message__groupId(const z_bmqa_Message* message_obj)
+{
+    using namespace BloombergLP;
+    const bmqa::Message* message_p = reinterpret_cast<const bmqa::Message*>(
+        message_obj);
+
+    return message_p->groupId().c_str();
+}
+
+#endif
+
+int z_bmqa_Message__messageGUID(const z_bmqa_Message*      message_obj,
+                                const z_bmqt_MessageGUID** messageGUID_obj)
+{
+    using namespace BloombergLP;
+    const bmqa::Message* message_p = reinterpret_cast<const bmqa::Message*>(
+        message_obj);
+    *messageGUID_obj = reinterpret_cast<const z_bmqt_MessageGUID*>(
+        &(message_p->messageGUID()));
+
+    return 0;
+}
+
+int z_bmqa_Message__confirmationCookie(
+    const z_bmqa_Message*              message_obj,
+    z_bmqa_MessageConfirmationCookie** cookie_obj)
+{
+    using namespace BloombergLP;
+    const bmqa::Message* message_p = reinterpret_cast<const bmqa::Message*>(
+        message_obj);
+    bmqa::MessageConfirmationCookie* cookie_p =
+        new bmqa::MessageConfirmationCookie(message_p->confirmationCookie());
+    *cookie_obj = reinterpret_cast<z_bmqa_MessageConfirmationCookie*>(
+        cookie_p);
+
+    return 0;
+}
+
+int z_bmqa_Message__ackStatus(const z_bmqa_Message* message_obj)
+{
+    using namespace BloombergLP;
+    const bmqa::Message* message_p = reinterpret_cast<const bmqa::Message*>(
+        message_obj);
+    return message_p->ackStatus();
+}
+
+// Add once we figure out how to handle Blobs in C
+// int z_bmqa_Message__getData(const z_bmqa_Message* message_obj, z_bdlbb_Blob*
+// blob) {
+//     using namespace BloombergLP;
+// const bmqa::Message* message_p = reinterpret_cast<const
+// bmqa::Message*>(message_obj);
+// }
+
+int z_bmqa_Message__getData(const z_bmqa_Message* message_obj, char** buffer)
+{
+    using namespace BloombergLP;
+    const bmqa::Message* message_p = reinterpret_cast<const bmqa::Message*>(
+        message_obj);
+
+    bdlbb::Blob       data;
+    int               rc = message_p->getData(&data);
+    bsl::stringstream ss;
+    ss << bdlbb::BlobUtilHexDumper(&data);
+    bsl::string data_str         = ss.str();
+    *buffer                      = new char[data_str.length() + 1];
+    (*buffer)[data_str.length()] = '\0';
+    strcpy(*buffer, data_str.c_str());
```

Comment:
While the transformation via HexDumper is  technically correct, it's got a few problems:
1. I'm pretty sure there is an unnecessary copy with the `stringstream::str` call
2. `strcpy` is an unsafe operation and should be avoided if possible
3. The buffer should probably be zero initialized

My preference would be to use [`bdlbb::BlobUtil::copy(char *dstBuffer, const Blob& srcBlob, int position, int length)`](https://bloomberg.github.io/bde-resources/doxygen/bde_api_prod/structbdlbb_1_1BlobUtil.html#afc81e01992b2a113ed612363becd13a8).

Please add a unit test to ensure that `z_bmqa_Message__getData()` works as intended.